app/src/main/java/com/kickstarter/services/KSApolloClientV2.kt

```diff
@@ -812,4 +815,33 @@ class KSApolloClientV2(val service: ApolloClient) : ApolloClientTypeV2 {
             return@defer ps
         }
     }
+
+    override fun fetchCategory(categoryParam: String): Observable<Category?> {
+        return Observable.defer {
+            val ps = PublishSubject.create<Category>()
+            this.service.query(
+                FetchCategoryQuery.builder()
+                    .categoryParam(categoryParam)
+                    .build()
+            ).enqueue(object : ApolloCall.Callback<FetchCategoryQuery.Data>() {
+                override fun onFailure(e: ApolloException) {
+                    ps.onError(e)
+                }
+
+                override fun onResponse(response: Response<FetchCategoryQuery.Data>) {
+                    response.data?.let { responseData ->
```

Comment:
I know we are moving this piece of code from the V1 version, but could we improve a bit from lines 832 to 840 with something like: 
```
if (response.hasErrors()) {
                            ps.onError(Exception(response.errors?.first()?.message))
                        } else {
                            val category = categoryTransformer(responseData.category()?.fragments()?.category())
                        ps.onNext(category)
                        }
                        ps.onComplete()
```
To observe for errored responses as well. 
(non-mandatory)